python-frontend/src/main/java/org/sonar/python/semantic/v2/types/Assignment.java

```diff
@@ -19,9 +19,86 @@
  */
 package org.sonar.python.semantic.v2.types;
 
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
 import org.sonar.plugins.python.api.tree.Expression;
 import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.Tree;
 import org.sonar.python.semantic.v2.SymbolV2;
+import org.sonar.python.semantic.v2.UsageV2;
+import org.sonar.python.tree.NameImpl;
+import org.sonar.python.types.HasTypeDependencies;
+import org.sonar.python.types.v2.PythonType;
+import org.sonar.python.types.v2.UnionType;
 
-public record Assignment(SymbolV2 lhsSymbol, Name lhsName, Expression rhs) {
+public class Assignment {
```

Comment:
do we really need to have it as a class instead of a record?

Reply:
I don't have a strong opinion on what it should be, but if you don't mind, I'd rather keep this as it is here, since SONARPY-1826 refactors this again. Please have a look at that PR and I'm happy to further refactor it there into a record if that still makes sense (I didn't consider this tbh).